cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricValue.java

```diff
@@ -44,10 +96,39 @@ public long getValue() {
 
   @Override
   public String toString() {
+    // TODO fix to handle distribution
     return "MetricValue{" +
       "name='" + name + '\'' +
       ", type=" + type +
       ", value=" + value +
       '}';
   }
+
+  public long[] getBucketCounts() {
+    return bucketCounts;
+  }
+
+  /**
+   * @return Counts for all buckets including buckets with zero counts.
+   * Helper function to publish the distribution metric to other systems such as Cloud Monitoring
+   */
+  public long[] getAllBucketCounts() {
+    return null;
```

Comment:
Why `return null`?

Reply:
This is temporary. It is a helper function. getBucketCounts will only return bucket counts of non-zero buckets. The caller has to consult the mask to see which buckets they stand for. getAllBucketCounts makes this easier. It looks at the mask and returns an array of size 64, with all buckets counts including zeroes.

Reply:
If it is for implementing it later, throw a `UnsupportedOperationException` instead of returning `null`.

Reply:
If it's a helper function, it's better to have it in the utility class. For serializable things additional getters may confuse serializers

Reply:
Throwing NotImplementedException. I think it is cleaner to have this method in this class. I think we have our own serialization code. Usually there is annotation to skip some methods during serialization. If that is not possible, I will create a helper class. Put a TODO to capture this.